pkg/openstack/loadbalancer.go

```diff
@@ -1340,6 +1340,38 @@ func (lbaas *LbaasV2) ensureOctaviaPool(lbID string, listener *listeners.Listene
 	if err != nil && err != openstackutil.ErrNotFound {
 		return nil, fmt.Errorf("error getting pool for listener %s: %v", listener.ID, err)
 	}
+
+	// By default, use the protocol of the listener
+	poolProto := v2pools.Protocol(listener.Protocol)
+	if svcConf.enableProxyProtocol {
+		poolProto = v2pools.ProtocolPROXY
+	} else if svcConf.keepClientIP && poolProto != v2pools.ProtocolHTTP {
```

Comment:
hmm, we should also remember that there can be udp and tcp loadbalancers.

Reply:
You are right, I updated the if clause to match the case inside `lbaas.buildPoolCreateOpt` now.

I also removed the log line because it would be printed later on too when the pool is getting created again.

However: I think udp and tcp is ok because we will get this information from the `listener.Protocol`.

What do you think?